packages/arb-token-bridge-ui/tests/e2e/specs/txHistory.cy.ts

```diff
@@ -0,0 +1,147 @@
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click()
+    cy.findByText(/page 2/i).should('be.visible')
+    cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should('have.length.above', 0)
+
+    // go to prev page
+    cy.findByRole('button', { name: /load previous deposits/i })
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click()
+    cy.findByText(/page 1/i).should('be.visible')
+    cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should('have.length.above', 0)
+  })
+
+  // Test deposits search works
+  it('should search deposits', () => {
+    // search for valid address substring
+    cy.findByPlaceholderText(DEPOSIT_SEARCH_IDENTIFIER)
+      .type('0x', { scrollBehavior: false })
+      .then(() => {
+        // wait for loader to appear before results
+        cy.wait(2000)
+
+        // search results
+        cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should(
+          'have.length.above',
+          0
+        )
+      })
+
+    // search for invalid address substring
+    cy.findByPlaceholderText(DEPOSIT_SEARCH_IDENTIFIER)
+      .clear()
+      .type('invalidTransactionHash', { scrollBehavior: false })
+      .then(() => {
+        cy.findByText(
+          /Oops! Looks like nothing matched your search query/i
+        ).should('be.visible')
+        cy.findByPlaceholderText(DEPOSIT_SEARCH_IDENTIFIER).clear()
+      })
+  })
+
+  // Test deposits withdrawal loading
+  it('should load withdrawals', () => {
+    cy.findByRole('tab', { name: 'show withdrawal transactions' })
+      .should('be.visible')
+      .click()
+      .should('have.class', 'selected')
+
+    cy.findAllByTestId(WITHDRAWAL_ROW_IDENTIFIER).should('have.length.above', 0)
+  })
+
+  // Test withdrawal pagination works
+  it('should paginate withdrawals', () => {
+    // check that initially previous button should be disabled
+    cy.findByRole('button', { name: /load previous withdrawals/i })
+      .should('be.visible')
+      .should('be.disabled')
+
+    // go to next page
+    cy.findByRole('button', { name: /load next withdrawals/i })
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click()
+    cy.findByText(/page 2/i).should('be.visible')
+    cy.findAllByTestId(WITHDRAWAL_ROW_IDENTIFIER).should('have.length.above', 0)
+
+    // go to prev page
+    cy.findByRole('button', { name: /load previous withdrawals/i })
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click()
+    cy.findByText(/page 1/i).should('be.visible')
+    cy.findAllByTestId(WITHDRAWAL_ROW_IDENTIFIER).should('have.length.above', 0)
+  })
+
+  // Test withdrawal search works
+  it('should search withdrawals', () => {
+    // search for valid address substring
+    cy.findByPlaceholderText(WITHDRAWAL_SEARCH_IDENTIFIER)
+      .type('0x', { scrollBehavior: false })
+      .then(() => {
+        // wait for loader to appear before results
+        cy.wait(2000)
```

Comment:
same as before, do we need this?

Reply:
Yes, same comment as before. If you try out the search UX and you will see why this wait made sense here.